ksql-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadata.java

```diff
@@ -34,6 +35,7 @@
 
 
   public PersistentQueryMetadata(final String statementString,
+                                 final StreamsConfig streamsConfig,
```

Comment:
Why does the streamsConfig need to be passed around? Its an internal config state. I can only see it being used in tests - in which case it shouldn't be passed around.

Reply:
Its just for testing. It was a style choice between making the streams config for a query readable or exposing the function that builds it. Now that I think about it a bit more it would be cleaner to just pull constructing the KafkaStreams out into its own interface/impl that I can mock.